### PR TITLE
🔀 Add user page

### PR DESCRIPTION
**Pull request by mileswong**
Connect #15
Connect #21

### CODE REVIEW

**Comment by inDream** [comment]
The user avatar should be generated in client side.

File: server/api/users.js [hunk 1]
```diff
@@ -23,6 +25,9 @@ router.get('/users/:id', async (req, res, next) => {
     const { id } = req.params;
     const user = await sequelize.user.findById(id, { raw: true });
     if (user) {
+      if (!user.avatar) {
+        user.avatar = toDataUrl(user.wallet);
```
Comment (inDream) [inline server/api/users.js]: The user avatar should be generated in client side.

File: util/mixin/masonry.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable sort-keys */
+import { sortImagesByHeight } from '~/util/masonry';
```
Comment (nwingt) [inline util/mixin/masonry.js]: I would suggest placing this in Masonry component and export as const

File: components/Masonry.vue [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    </div>
+  </div>
+</template>
+
+<script>
+import LikeButton from '~/components/LikeButton';
+import UserBadge from '~/components/UserBadge';
+
+export default {
+  name: 'masonry',
```
Comment (nwingt) [inline components/Masonry.vue]: Prefer `masonry-images-grid` for fulfilling https://vuejs.org/v2/style-guide/#Multi-word-component-names-essential

File: components/UserBadge.vue [hunk 4]
```diff
@@ -57,6 +58,12 @@ export default {
     },
   },
   computed: {
+    avatar() {
+      return (
+        this.user.avatar ||
+        (this.user.wallet ? toDataUrl(this.user.wallet) : null)
```
Comment (inDream) [inline components/UserBadge.vue]: `this.user.wallet` should not be optional

File: util/mixin/masonryImagesGrid.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        colCount = 2;
+      }
+      if (colCount !== this.colCount) {
+        this.colCount = colCount;
+        this.images = sortImagesByHeight(this.rawImages, colCount);
+      }
+    },
+    sortImagesByHeight,
+  },
+};
```
Comment (nwingt) [inline util/mixin/masonryImagesGrid.js]: Please put the entire mixin within the component and export as const

File: components/MasonryImagesGrid.vue [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+  </div>
+</template>
+
+<script>
+import LikeButton from '~/components/LikeButton';
+import UserBadge from '~/components/UserBadge';
+
+const MIN_HEIGHT = 180;
+
+export const sortImagesByHeight = (images, colCount) => {
```
Comment (nwingt) [inline components/MasonryImagesGrid.vue]: No need to export

**Pull request merged**